providers/shared/network.go

```diff
@@ -29,15 +29,19 @@ func Network() (ips, macs []string, err error) {
 
 	ips = make([]string, 0, len(ifcs))
```

Comment:
I think we should move the allocations closer to where they are used and base their capacity on the size of the list we are iterating over.

```diff
        }
 
-       ips = make([]string, 0, len(ifcs))
-       macs = make([]string, 0, len(ifcs))
-
        // This function fetches all the addresses in a single syscall. Fetching addresses individually for each interface
        // can be expensive when the host has a lot of interfaces. This usually happens when the host is doing virtualized
        // networking for guests, in Kubernetes for example.
@@ -37,10 +34,12 @@ func Network() (ips, macs []string, err error) {
        if err != nil {
                return nil, nil, err
        }
+       ips = make([]string, 0, len(addrs))
        for _, addr := range addrs {
                ips = append(ips, addr.String())
        }
 
+       macs = make([]string, 0, len(ifcs))
        for _, ifc := range ifcs {
                mac := ifc.HardwareAddr.String()
```